var/spack/repos/spack_repo/builtin/packages/root/package.py

```diff
@@ -401,11 +430,11 @@ class Root(CMakePackage):
     depends_on("graphviz", when="+graphviz")
     depends_on("gsl", when="+gsl")
     depends_on("jemalloc", when="+jemalloc")
-    depends_on("mysql-client", when="+mysql")
+    depends_on("mysql-client", when="+mysql @:6.36")
```

Comment:
When is implicit in variant.
```suggestion
    depends_on("mysql-client", when="+mysql")
```